src/nupic/engine/LinkPolicyFactory.cpp

```diff
@@ -30,7 +30,7 @@ namespace nupic {
 
 LinkPolicy *LinkPolicyFactory::createLinkPolicy(const std::string policyType,
                                                 const std::string policyParams,
-                                                Link *link) {
+                                                Link* link) {
   LinkPolicy *lp = nullptr;
```

Comment:
No, a unique_ptr is to hold a pointer that should be deleted when it goes out of scope.  The link policy just needs to get some things from the link.  (probably should be a const pointer however.)
There are other places where the link pointer is held in more than one containor...so it may be apporpreate for a shared_ptr.